sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -3020,6 +3020,14 @@ class DataFrameSuite extends QueryTest
       }
     }
   }
+
+  test("SPARK-36488: Regular expression expansion should fail with a meaningful message") {
```

Comment:
This test should be added in `AnalysisErrorSuite`. Also, could you add another test for invalid star usage when `SUPPORT_QUOTED_REGEX_COLUMN_NAME` is true?